sys/net/gnrc/network_layer/sixlowpan/iphc/gnrc_sixlowpan_iphc.c

```diff
@@ -562,8 +562,21 @@ inline static size_t iphc_nhc_udp_encode(gnrc_pktsnip_t *udp, ipv6_hdr_t *ipv6_h
 
     /* Set UDP header ID (rfc6282#section-5). */
     ipv6_hdr->nh |= NHC_UDP_ID;
-    /* shrink udp allocation to final size */
-    gnrc_pktbuf_realloc_data(udp, nhc_len);
+
+    if (udp->type == GNRC_NETTYPE_IPV6) {
+        /* forwarded ipv6 packet */
+        size_t diff = sizeof(udp_hdr_t) - nhc_len;
+        for (int i = nhc_len; i < (udp->size - diff); i++) {
```

Comment:
Well, it's not really broken but it forces NHC to be included by default, on which Murdock fails as the variables are not of the same type. But for me a `size_t i = ...` it somehow weird though... But well, it should be fixed somehow.